partner_communication_revision/data/demo.xml

```diff
@@ -0,0 +1,39 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+    <data>
+        <record id="template" model="mail.template">
+            <field name="name">Donation - Thank You Letter</field>
+            <field name="model_id" ref="partner_communication.model_partner_communication_job"/>
+            <field name="use_default_to" eval="True"/>
+            <field name="subject">Donation Receipt</field>
+            <field name="body_html" type="html">
+                <div>
+                    % set invoice_lines = object.get_objects()
```

Comment:
get_objects() is a method in `partner.communication.job` that returns the related objects. In this case, since in your config you put the related model as `res_partner`, the method will return a `res.partner` recordset. Since you don't use invoice_lines later in the template, it's not a real problem but you could rename this variable for better clarity.